applications/SwimmingDEMApplication/tests/SmallTests.py

```diff
@@ -67,14 +75,17 @@ class candelier_with_history_hinsberg_test(CandelierTF.TestFactory):
      #      file_name = "candelier_tests/candelier"
      #      file_parameters = "candelier_tests/ProjectParametersWithHistoryNonInertial.json"
 
-     available_tests += [test_class for test_class in CandelierTF.TestFactory.__subclasses__()]
 
 if fluid_DEM_coupling_imports_available:
      class fluid_dem_coupling_one_way_test(FDEMTF.TestFactory):
           file_name = "fluid_dem_tests/settling_cube"
           file_parameters = "fluid_dem_tests/ProjectParameters.json"
 
-     available_tests += [test_class for test_class in FDEMTF.TestFactory.__subclasses__()]
+available_tests = []
+available_tests += [test_class for test_class in InterpolationTF.TestFactory.__subclasses__()]
+available_tests += [test_class for test_class in BackwardCouplingTF.TestFactory.__subclasses__()]
+available_tests += [test_class for test_class in CandelierTF.TestFactory.__subclasses__()]
```

Comment:
I think these lines have to be inside the conditions you set to define the subclasses. In fact, you are not able to import the CandelierTF, so your code goes to line 22 and it does not define the classes of the candelier tests,but you are appending in that line, so the code crashes. You should unchange this approach.

Reply:
But if the import was not possible, then there are no subclasses appended. I do not see the problem.

Reply:
Oh, yes. You are right.